lib/ruby-enum/enum.rb

```diff
@@ -118,7 +118,10 @@ def keys
 
       # Returns all enum values.
       def values
-        @_enum_hash.values.map(&:value)
+        values = @_enum_hash.values.map(&:value)
+        superclass.values + values
+      rescue NoMethodError
```

Comment:
I don't love relying on an exception for control here, is there a reason not to type check?

Reply:
No. No good reason really. It's just a paradigm brought from Python: "leap before you look".

Reply:
I replaced this.